flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/reader/deserializer/PulsarSchemaWrapper.java

```diff
@@ -57,16 +57,20 @@ public void open(InitializationContext context) throws Exception {
     }
 
     @Override
-    public void deserialize(Message<byte[]> message, Collector<T> out) throws Exception {
-        byte[] bytes = message.getData();
-        T instance = schema.decode(bytes);
-
-        out.collect(instance);
+    public void deserialize(Message<?> message, Collector<T> out) throws Exception {
+        out.collect((T) message.getValue());
```

Comment:
here we do just get the value from the decoded message. One question is , for key and value type, do we retrieve both key and value ? How is that done ?

Reply:
This implementation is not good at all.

Reply:
I think instead of just saying it's not good, it might be better to point out directly why it's not good so I can make it better : )